deepinv/sampling/langevin.py

```diff
@@ -256,6 +31,8 @@ class ULA(MonteCarlo):
     - Projected PnP-ULA assumes that the denoiser is :math:`L`-Lipschitz differentiable
     - For convergence, ULA required step_size smaller than :math:`\frac{1}{L+\|A\|_2^2}`
 
+    .. warning::
+        This is a legacy class provided for convenience. See :ref:`sphx_glr_auto_examples_sampling_demo_sampling.py` for more details on sampling implementation, and how to adapt it to your specific problem.
```

Comment:
I would reword this to suggest that you can still use it, but you can get more flexibility by simply building the ULA sampler yourself, and to see the example in the User Guide on how to do this. (I think referecing the quite complicated example might put users off).

Same with skrock below.